okhttp-tls/src/main/java/okhttp3/tls/HeldCertificate.kt

```diff
@@ -111,52 +111,90 @@ import javax.security.auth.x500.X500Principal
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "certificate"),
+      level = DeprecationLevel.WARNING)
   fun certificate(): X509Certificate = certificate
 
+  @JvmName("-deprecated_keyPair")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "keyPair"),
+      level = DeprecationLevel.WARNING)
   fun keyPair(): KeyPair = keyPair
 
   /**
    * Returns the certificate encoded in [PEM format][rfc_7468].
    *
    * [rfc_7468]: https://tools.ietf.org/html/rfc7468
    */
-  fun certificatePem(): String {
-    return buildString {
-      append("-----BEGIN CERTIFICATE-----\n")
-      encodeBase64Lines(certificate.encoded.toByteString())
-      append("-----END CERTIFICATE-----\n")
+  @get:JvmName("certificatePem") val certificatePem: String
+    get() {
+      return buildString {
+        append("-----BEGIN CERTIFICATE-----\n")
+        encodeBase64Lines(certificate.encoded.toByteString())
+        append("-----END CERTIFICATE-----\n")
+      }
     }
-  }
+
+  @JvmName("-deprecated_certificatePem")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "certificatePem"),
+      level = DeprecationLevel.WARNING)
+  fun certificatePem(): String = certificatePem
 
   /**
    * Returns the RSA private key encoded in [PKCS #8][rfc_5208] [PEM format][rfc_7468].
    *
    * [rfc_5208]: https://tools.ietf.org/html/rfc5208
    * [rfc_7468]: https://tools.ietf.org/html/rfc7468
    */
-  fun privateKeyPkcs8Pem(): String {
-    return buildString {
-      append("-----BEGIN PRIVATE KEY-----\n")
-      encodeBase64Lines(keyPair.private.encoded.toByteString())
-      append("-----END PRIVATE KEY-----\n")
+  @get:JvmName("privateKeyPkcs8Pem") val privateKeyPkcs8Pem: String
+    get() {
+      return buildString {
+        append("-----BEGIN PRIVATE KEY-----\n")
+        encodeBase64Lines(keyPair.private.encoded.toByteString())
+        append("-----END PRIVATE KEY-----\n")
+      }
     }
-  }
+
+  @JvmName("-deprecated_privateKeyPkcs8Pem")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "privateKeyPkcs8Pem"),
+      level = DeprecationLevel.WARNING)
+  fun privateKeyPkcs8Pem(): String = privateKeyPkcs8Pem
 
   /**
    * Returns the RSA private key encoded in [PKCS #1][rfc_8017] [PEM format][rfc_7468].
    *
    * [rfc_8017]: https://tools.ietf.org/html/rfc8017
    * [rfc_7468]: https://tools.ietf.org/html/rfc7468
    */
-  fun privateKeyPkcs1Pem(): String {
-    require(keyPair.private is RSAPrivateKey) { "PKCS1 only supports RSA keys" }
-    return buildString {
-      append("-----BEGIN RSA PRIVATE KEY-----\n")
-      encodeBase64Lines(pkcs1Bytes())
-      append("-----END RSA PRIVATE KEY-----\n")
+  @get:JvmName("privateKeyPkcs1Pem") val privateKeyPkcs1Pem: String
+    get() {
+      require(keyPair.private is RSAPrivateKey) { "PKCS1 only supports RSA keys" }
```

Comment:
should a method that has a require internally ever be a val?

Reply:
Should probably be `check`. The original function seems wrong. How can you throw illegal argument when you accept no arguments?

Reply:
Thanks! I replaced `require` with `check` and intentionally split the change into another commit since it's not related to idiomatic Kotlin change. (if it's okay and makes sense :) )